eclair-core/src/main/scala/fr/acinq/eclair/db/migration/CompareDb.scala

```diff
@@ -0,0 +1,80 @@
+    while (rs2.next()) {
+      hashes2 = hash2(rs2) +: hashes2
+    }
+
+    val res = hashes1.sorted == hashes2.sorted
+    val diff1 = hashes1 diff hashes2
+    val diff2 = hashes2 diff hashes1
+
+    if (res) {
+      logger.info(s"tables $table1/$table2 are identical")
+    } else {
+      logger.warn(s"tables $table1/$table2 are different diff1=${diff1.take(3).map(_.toHex.take(128))} diff2=${diff2.take(3).map(_.toHex.take(128))}")
+    }
+
+    res
+  }
+
+  // @formatter:off
+  import fr.acinq.eclair.db.jdbc.JdbcUtils.ExtendedResultSet._
+  def bytes(rs: ResultSet, columnName: String): ByteVector = rs.getByteVector(columnName)
+  def bytesnull(rs: ResultSet, columnName: String): ByteVector = rs.getByteVectorNullable(columnName).getOrElse(ByteVector.fromValidHex("deadbeef"))
+  def hex(rs: ResultSet, columnName: String): ByteVector = rs.getByteVectorFromHex(columnName)
+  def hexnull(rs: ResultSet, columnName: String): ByteVector = rs.getByteVector32FromHexNullable(columnName).map(_.bytes).getOrElse(ByteVector.fromValidHex("deadbeef"))
+  def string(rs: ResultSet, columnName: String): ByteVector = ByteVector(rs.getString(columnName).getBytes)
+  def stringnull(rs: ResultSet, columnName: String): ByteVector = ByteVector(rs.getStringNullable(columnName).getOrElse("<null>").getBytes)
+  def bool(rs: ResultSet, columnName: String): ByteVector = ByteVector.fromByte(if (rs.getBoolean(columnName)) 1 else 0)
+  def long(rs: ResultSet, columnName: String): ByteVector = ByteVector.fromLong(rs.getLong(columnName))
+  def longnull(rs: ResultSet, columnName: String): ByteVector = ByteVector.fromLong(rs.getLongNullable(columnName).getOrElse(42))
+  def longts(rs: ResultSet, columnName: String): ByteVector = ByteVector.fromLong((rs.getLong(columnName).toDouble / 1_000_000).round)
+  def longtsnull(rs: ResultSet, columnName: String): ByteVector = ByteVector.fromLong(rs.getLongNullable(columnName).map(l => (l.toDouble/1_000_000).round).getOrElse(42))
+  def int(rs: ResultSet, columnName: String): ByteVector = ByteVector.fromInt(rs.getInt(columnName))
+  def ts(rs: ResultSet, columnName: String): ByteVector = ByteVector.fromLong((rs.getTimestamp(columnName).getTime.toDouble / 1_000_000).round)
+  def tsnull(rs: ResultSet, columnName: String): ByteVector = ByteVector.fromLong(rs.getTimestampNullable(columnName).map(t => (t.getTime.toDouble / 1_000_000).round).getOrElse(42))
+  def tssec(rs: ResultSet, columnName: String): ByteVector = ByteVector.fromLong((rs.getTimestamp(columnName).toInstant.getEpochSecond.toDouble / 1_000_000).round)
+  def tssecnull(rs: ResultSet, columnName: String): ByteVector = ByteVector.fromLong(rs.getTimestampNullable(columnName).map(t => (t.toInstant.getEpochSecond.toDouble / 1_000_000).round).getOrElse(42))
+  // @formatter:on
+
+  def compareAll(dualDatabases: DualDatabases): Unit = {
+    logger.info("comparing all tables...")
+    val (sqliteDb, postgresDb) = DualDatabases.getDatabases(dualDatabases)
```

Comment:
This assumes that `sqlite` is primary, all these comparisons won't work if postgres is primary (because inside the comparisons you assign the table names so that the first one is sqlite...)

Reply:
Nice catch, fixed with 5c3eac546. I've updated `Databases.init` so we only migrate/compare when sqlite is primary, and I added explicit casts in `migrateAll`/`compareAll`.